packages/authentication/test/unit/authentication.ts

```diff
@@ -0,0 +1,90 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from 'testlab';
+import {Authentication} from '../../src/authentication';
+import {BasicStrategy} from 'passport-http';
+const http = require('http');
+
+describe('Authentication', () => {
+  let passport;
+  let strategy;
+  let req;
+  let res;
+
+  describe('Basic Strategy', () => {
+    beforeEach(createBasicStrategy);
+    beforeEach(createPassport);
+    beforeEach(createMockHttpObjects);
+
+    it('authenticates successfully for correct credentials', () => {
```

Comment:
This test is invoking an async method `passport.authenticate` but does not wait until that method is finished. This typically causes the assertion error thrown by this test to fail one of the next tests that happen to be running at the time when the assertion error was thrown.

Have you seen this test to fail?

Here is a possible fix:

```ts
it('...', done => {
  // ...
  passport.authenticate(req, res, err => {
    if (err) return done(err);
    expect(req.user).to.have.property('role', 'admin');
  }
```